drivers/place/event_mailer.cr

```diff
@@ -0,0 +1,144 @@
+  def list_systems_in_zone(zone_id : String)
+    staff_api.systems(zone_id: zone_id).get.as_a
+  end
+
+  private def process_updated_events(system_id : String, events : Array(PlaceCalendar::Event))
+    selected_events = apply_filter(events)
+    new_events = if @events[system_id]
+                   @events[system_id] - selected_events
+                 else
+                   selected_events
+                 end
+    @events[system_id] = selected_events # Store the updated list of events
+    new_events.each { |event| send_event_email(event, system_id) }
+  end
+
+  private def send_event_email(event : PlaceCalendar::Event, system_id : String)
+    return "Event email was already sent at #{event.extension_data[:event_mailer_email_sent_at]}" if event.extension_data[:event_mailer_email_sent_at]
+    organizer_email = event.host
+    organizer_name = event.attendees.find { |a| a.email == organizer_email }.try &.name || "Name Unknown"
+    network_username = network_password = ""
+    network_username, network_password = fetch_network_user(organizer_email.not_nil!, organizer_name.not_nil!) if @send_network_credentials
+
+    email_data = {
+      host_name:        organizer_name,
+      host_email:       organizer_email,
+      room_name:        event.location,
+      event_title:      event.title,
+      event_start:      event.event_start.to_s(@time_format),
+      event_date:       event.event_end.not_nil!.to_s(@date_format),
+      network_username: network_username,
+      network_password: network_password,
+    }
+    mailer.send_template(
+      to: [organizer_email],
+      template: {"events", @send_network_credentials ? "welcome_with_network_credentials" : "welcome"},
+      args: email_data
+    )
+    staff_api.patch_event_metadata(system_id, event.id, {"event_mailer_email_sent_at": Time.local}.to_json).get
+  end
+
+  private def apply_filter(events : Array(PlaceCalendar::Event))
+    # Additional event filters can be added in the future
+    case @event_filter
+    when :occurs_today
+      logger.debug { "Event filter: occurs today" } if @debug
+      select_todays_events(events)
+    else
+      logger.debug { "Event filter: NONE" } if @debug
+      events
+    end
+  end
+
+  private def select_todays_events(events : Array(PlaceCalendar::Event))
+    events.select do |event|
+      timezone = event.timezone ? Time::Location.load(event.timezone.not_nil!) : Time::Location.local
+      now = Time.local(location: timezone)
+      event.event_start >= now.at_beginning_of_day && event.event_start <= now.at_end_of_day
+    end
+  end
+
+  private def fetch_network_user(user_email : String, user_name : String)
+    # Check if they already exist
+    response = JSON.parse(network_provider.get_internal_user_by_email(user_email).get.as_s)
+    logger.debug { "Response from Network Identity provider for lookup of #{user_email} was:\n #{response} } " } if @debug
+    return {response["name"], response["password"]} unless response["error"]
```

Comment:
You need to update the password for the returned user since the password is returned as ***** and not clear-text.

Reply:
Use [this](https://github.com/PlaceOS/drivers/blob/master/drivers/cisco/ise/guests.cr#L188) or [this](https://github.com/PlaceOS/drivers/blob/master/drivers/cisco/ise/guests.cr#L202) to update the password of an already existing user.